### ISSUE TITLE
sortedImports borkage

### ISSUE THREAD

**Issue by gurgeous** [nicklockwood/SwiftFormat#698]
Hey Nick, thanks for SwiftFormat and your many contributions to #ioslife... We couldn't build apps without you!

There's a wee bug in sortedImports, triggered by one of our `SwifterSwift` files. I narrowed it down with a small test case. Something to do with the header comment, the nested canImports, and `sortedImports. If I remove the comment the bug no longer occurs.

before

```swift
// evil comment

#if canImport(Foundation)
    import Foundation
    #if canImport(UIKit) && canImport(AVFoundation)
        import UIKit
        import AVFoundation
    #endif
#endif
```

`$ swiftformat . --swiftversion 5`

after

```swift
import AVFoundation
// evil comment

#if canImport(Foundation)
    import Foundation
    #if canImport(UIKit) && canImport(AVFoundation)
        import UIKit
    #endif
#endif
```

**Comment by nicklockwood** [comment]
@gurgeous thanks for reporting, I'll try to fix it asap

**Comment by nicklockwood** [comment]
@gurgeous fixed in 0.45.5.
